File: scripts/generate_android_icons.py
#!/usr/bin/env python3
import os
import sys
from PIL import Image, ImageOps, ImageDraw

# Paths
ROOT_DIR = "d:/PRAHARI_complete-1/PRAHARI"
RES_DIR = os.path.join(ROOT_DIR, "android/app/src/main/res")
SOURCE_IMG = "C:/Users/KARAN KUMAR JAUHAR/.gemini/antigravity/brain/346ce96d-8f24-4b40-9ac3-1b3db7561991/prahari_app_icon_1780448904598.png"

# Icon dimensions for Android launcher
SIZES = {
    "mipmap-mdpi": 48,
    "mipmap-hdpi": 72,
    "mipmap-xhdpi": 96,
    "mipmap-xxhdpi": 144,
    "mipmap-xxxhdpi": 192
}

def make_circular(img):
    """Applies a high-quality circular mask to the image."""
    size = img.size
    mask = Image.new('L', size, 0)
    draw = ImageDraw.Draw(mask)
    draw.ellipse((0, 0) + size, fill=255)
    
    output = Image.new('RGBA', size, (0, 0, 0, 0))
    output.paste(img, (0, 0), mask=mask)
    return output

def process_icons():
    if not os.path.exists(SOURCE_IMG):
        print(f"Error: Source image not found at {SOURCE_IMG}")
        sys.exit(1)
        
    try:
        base_img = Image.open(SOURCE_IMG).convert("RGBA")
    except Exception as e:
        print(f"Error opening source image: {e}")
        sys.exit(1)
        
    print(f"Processing icons from source: {SOURCE_IMG}")
    
    for folder, size in SIZES.items():
        folder_path = os.path.join(RES_DIR, folder)
        os.makedirs(folder_path, exist_ok=True)
        
        # Resize using high-quality Lanczos resampling
        resized_square = base_img.resize((size, size), Image.Resampling.LANCZOS)
        
        # Save square icon (ic_launcher.png)
        square_path = os.path.join(folder_path, "ic_launcher.png")
        resized_square.save(square_path, "PNG", optimize=True)
        print(f"Saved {square_path} ({size}x{size})")
        
        # Save round icon (ic_launcher_round.png)
        resized_round = make_circular(resized_square)
        round_path = os.path.join(folder_path, "ic_launcher_round.png")
        resized_round.save(round_path, "PNG", optimize=True)
        print(f"Saved {round_path} ({size}x{size})")
        
    print("Successfully generated and optimized all Android Launcher Icons!")

if __name__ == "__main__":
    process_icons()

File: scripts/generate_android_icons.js
// scripts/generate_android_icons.js
// Node.js script to resize and mask app icons for Android using 'jimp'.
// Pure JavaScript image library with zero native compiling dependencies.

const fs = require('fs');
const path = require('path');

async function run() {
  let Jimp;
  try {
    const jimpModule = require('jimp');
    Jimp = jimpModule.Jimp || jimpModule;
  } catch (err) {
    console.error('Error: "jimp" is not installed. Please run "npm install jimp" first.');
    process.exit(1);
  }

  const sourceImgPath = "C:\\Users\\KARAN KUMAR JAUHAR\\.gemini\\antigravity\\brain\\346ce96d-8f24-4b40-9ac3-1b3db7561991\\prahari_high_visibility_icon_1780452215001.png";
  const resDir = "d:\\PRAHARI_complete-1\\PRAHARI\\android\\app\\src\\main\\res";

  // Android Launcher Icon sizes
  const sizes = {
    "mipmap-mdpi": 48,
    "mipmap-hdpi": 72,
    "mipmap-xhdpi": 96,
    "mipmap-xxhdpi": 144,
    "mipmap-xxxhdpi": 192
  };

  if (!fs.existsSync(sourceImgPath)) {
    console.error(`Error: Source image not found at: ${sourceImgPath}`);
    process.exit(1);
  }

  console.log(`Loading source image: ${sourceImgPath}`);

  for (const [folder, size] of Object.entries(sizes)) {
    const folderPath = path.join(resDir, folder);
    if (!fs.existsSync(folderPath)) {
      fs.mkdirSync(folderPath, { recursive: true });
    }

    try {
      // 1. Generate Square Icon (ic_launcher.png)
      const image = await Jimp.read(sourceImgPath);
      
      // Resize with high quality
      image.resize({ w: size, h: size });
      
      // Save square icon
      const squarePath = path.join(folderPath, "ic_launcher.png");
      await image.write(squarePath);
      console.log(`Saved: ${squarePath} (${size}x${size})`);

      // 2. Generate Round Icon (ic_launcher_round.png)
      // Custom pixel loop to mask to circle (extremely robust, pure JS)
      const radius = size / 2;
      const center = radius;
      
      image.scan(0, 0, size, size, function (x, y, idx) {
        // Calculate distance from center of the pixel to the center of the image
        const distance = Math.sqrt(Math.pow(x - center, 2) + Math.pow(y - center, 2));
        if (distance > radius) {
          this.bitmap.data[idx + 3] = 0; // Set alpha channel to 0 (fully transparent)
        }
      });

      const roundPath = path.join(folderPath, "ic_launcher_round.png");
      await image.write(roundPath);
      console.log(`Saved: ${roundPath} (${size}x${size})`);

    } catch (iconError) {
      console.error(`Error generating icons for size ${size}:`, iconError);
    }
  }

  console.log("Success: All Android launcher icons generated successfully using Jimp!");
}

run();
